crates/rome_json_formatter/src/json/lists/array_element_list.rs

```diff
@@ -1,12 +1,105 @@
 use crate::prelude::*;
+use crate::separated::FormatAstSeparatedListExtension;
+use rome_formatter::write;
 use rome_json_syntax::JsonArrayElementList;
+use rome_rowan::{AstNode, AstSeparatedList};
 
 #[derive(Debug, Clone, Default)]
 pub(crate) struct FormatJsonArrayElementList;
 
 impl FormatRule<JsonArrayElementList> for FormatJsonArrayElementList {
     type Context = JsonFormatContext;
     fn fmt(&self, node: &JsonArrayElementList, f: &mut JsonFormatter) -> FormatResult<()> {
-        format_verbatim_node(node.syntax()).fmt(f)
+        let layout = if can_concisely_print_array_list(node) {
+            ArrayLayout::Fill
+        } else {
+            ArrayLayout::OnePerLine
+        };
+
+        match layout {
+            ArrayLayout::Fill => {
+                let mut filler = f.fill();
+
+                // Using format_separated is valid in this case as can_print_fill does not allow holes
+                for (element, formatted) in node.iter().zip(node.format_separated(",")) {
+                    filler.entry(
+                        &format_once(|f| {
+                            if get_lines_before(element?.syntax()) > 1 {
+                                write!(f, [empty_line()])
+                            } else {
+                                write!(f, [soft_line_break_or_space()])
+                            }
+                        }),
+                        &formatted,
+                    );
+                }
+
+                filler.finish()
+            }
+
+            ArrayLayout::OnePerLine => {
+                let mut join = f.join_nodes_with_soft_line();
+
+                for (element, formatted) in node.elements().zip(node.format_separated(",")) {
+                    join.entry(element.node()?.syntax(), &formatted);
+                }
+
+                join.finish()
+            }
+        }
     }
 }
+
+#[derive(Copy, Clone, Debug)]
+enum ArrayLayout {
+    /// Tries to fit as many array elements on a single line as possible.
+    ///
+    /// ```json
+    /// { list: [
+    ///     1, 2, 3,
+    ///     5, 6,
+    ///   ]
+    /// }
```

Comment:
This is not valid JSON. ou can simplify it to 
```suggestion
    /// [
    ///     1, 2, 3,
    ///     5, 6,
    /// ]
```